recipe/meta.yaml

```diff
@@ -1,33 +1,33 @@
-{% set version = "0.9.8" %}
+{% set version = "1.2.0" %}
 
 package:
   name: requests-cache
   version: {{ version }}
 
 source:
   url: https://pypi.org/packages/source/r/requests-cache/requests_cache-{{ version }}.tar.gz
-  sha256: eaed4eb5fd5c392ba5e7cfa000d4ab96b1d32c1a1620f37aa558c43741ac362b
+  sha256: db1c709ca343cc1cd5b6c8b1a5387298eceed02306a6040760db538c885e3838
 
 build:
-  noarch: python
   number: 0
-  script: {{ PYTHON }} -m pip install . -vv
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  skip: True  # [py<38]
 
 requirements:
   host:
-    - python >=3.7
-    - poetry
+    - python
+    - poetry-core
     - pip
 
   run:
-    - python >=3.7
-    - appdirs >=1.4.4
+    - python
```

Comment:
from [here](https://github.com/requests-cache/requests-cache/blob/c71837676ccfe4c471030d13289c187a2560a51f/pyproject.toml#L61) you've got all but one of the serialization dependencies, is there a reason for that?

Reply:
I would move all extra/optional dependencies to `run_constrained` if they have pinnings.

Reply:
Each are independent if I understand it correctly. For consistency with past releases, optional dependencies that were in our run section should be left there. And others added to run_constrained.

Reply:
> from [here](https://github.com/requests-cache/requests-cache/blob/c71837676ccfe4c471030d13289c187a2560a51f/pyproject.toml#L61) you've got all but one of the serialization dependencies, is there a reason for that?

Yes, all the other optional dependencies already existed and `bson`, the only optional dependency missing is a dependency for the optional backend dependency `pymongo`, so I thought it was safe to leave it out.

Reply:
> I would move all extra/optional dependencies to `run_constrained` if they have pinnings.

Is it okay to not add to `run_constrained` if it's verified that we have the minimal requirements for each or should this be done for every package regardless if we have it or not?

I'm still learning about how `run_constrained` works.

Reply:
> Each are independent if I understand it correctly. For consistency with past releases, optional dependencies that were in our run section should be left there. And others added to run_constrained.

I'm just seeing this. I believe so as well. I only updated the optional dependencies already listed and added the required `platformdirs`.

Reply:
I think we're pretty loose about whether things should be added to run_constrained or not. It's to pin optional dependencies. i.e., to say, "if this package is somewhere in the environment as well, then it _must_ be of a certain version range". However the optional deps upstream may not correspond exactly to our run_constrained because:
1. we consider the feature the dep supports to be important, so we make it non-optional and put them in `run`
2. there's no pinning specified upstream so there's no point in putting them in run_constrained

Reply:
Given upstream specified constraints, we should add these:
https://github.com/requests-cache/requests-cache/blob/v1.2.0/pyproject.toml#L56-L62
```
boto3 >=1.15
botocore >=1.18
pymongo >=3
redis-py >=3
bson >=0.5
```

Reply:
Well, in this case, it seemed important to update those optional dependencies since those particular dependencies were already in the recipe, so I left them in the `run` section, where they already existed. I don't mind adding the other optional backend dependencies, but `bson` does not exist on the `main` channel and `pymongo` which has not been updated for three years depends on it. My analysis of our options are as follows: (Please correct me if this way of thinking is incorrect)

1. Add the last optional serialization dependency (`bson`) and all of the optional backend dependencies (Means rebuilding `pymongo` and adding `bson` to defaults`. Also, means discussion on renaming/revamping `pymongo` due to it being severely out of date.)
2. Remove all optional dependencies, including previously listed ones (Means if a customer was used to having said dependency already bundled with this package, it may cause problems if removed now.) 
3. Leave as is (Means the dependencies already bundled will remain and we possibly have an opportunity to earn credits for the extra effort to include the other optional dependencies.)

What would be the best course of action from here?

Reply:
It is fine to add run_constrained for dependencies we don't have in main. If a user tries to install requests-cache with pymongo, the run_constrained check will prevent it. Also, users may use additional channels which provide those packages.